indico/modules/events/registration/client/js/form/FormItem.jsx

```diff
@@ -145,11 +183,22 @@ export default function FormItem({
           disabled={disabled}
           isPurged={showPurged}
           htmlId={htmlId}
+          htmlName={htmlName}
           {...inputProps}
         />
       </>
     ) : null;
 
+  useEffect(() => {
+    if (!show && !setupMode) {
+      form.change(htmlName, undefined);
```

Comment:
We also need to handle the case where a field becomes visible: If it has a default value, we need to set the value to that one.

Reply:
Also, when editing a registration and toggling the condition field, then we should probably use the existing data instead of the initial data so it doesn't just disappear.

Reply:
> If it has a default value

And if it has, how can that value be obtained? I haven't seen anything related to field's default values in the registry.

Reply:
`const {initialValues} = useSelector(getStaticData);`

this is also what's used in the `RegistrationFormSubmission` component to populate the initial form data

Reply:
There is actually a `defaultValue` property being passed. I used that one in [50ca2ec](https://github.com/indico/indico/pull/6678/commits/50ca2ecda55efcebc5f491a3e9dc84522e1ef770).